### PR TITLE
Feature/extra class hooks

### PR DESCRIPTION
**Pull request by CSergienko**
# Background
In the process of implementing RWC and having to restyle the defaults, I came across a few situations where there were certain hooks I needed that were missing.

The changes in this PR are backwards compatible.

# What was done
For one, I noticed I was intermittently getting an error due to `message.length` being undefined, causing the entire thing to explode. As such, I have added a guard to the action where this particular piece of data was being used.

I added an `is-local` class to `MessageList-item` containers, which allow the implementer to now be able to style bot and user bubbles differently from a higher point in the tree.

I also added an `is-expected-input` class to the input at the bottom of the UI, which is a useful hook for when the implementer would like to hide the 

I have also corrected a markup issue where the way our messages were being wrapped looked like this: `ul > div > li`

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-lts/carbon
+8.12.0
```
Comment (chrislombaard) [inline .nvmrc]: Is there a specific reason we're not switching to node v10?

File: src/components/MessageList/index.js [hunk 2]
```diff
@@ -51,14 +51,16 @@ class MessageList extends React.Component {
...
                 className="ChatContainer-content"
                 ref={ref => (this._ref = ref)}
             >
-                <div className="MessagesList">
+                <ul className="MessagesList">
                     {messages.map((message, i) => (
                         <li
-                            className="MessagesList-item"
+                            className={`MessagesList-item${
+                                message.origin === 'local' ? ' is-local' : ''
```
Comment (chrislombaard) [inline src/components/MessageList/index.js]: What does `is-local` mean?

File: .nvmrc (hunk 1 above)
Comment (CSergienko) [reply .nvmrc]: We're sticking with the LTS version for the sake of stability.

File: src/components/MessageList/index.js (hunk 2 above)
Comment (CSergienko) [reply src/components/MessageList/index.js]: This is just a class on the element. It's a convention being used elsewhere. Basically, it means the source of the message is from the user and not the bot.

**Pull request closed without merging**